examples/3d/render_to_texture.rs

```diff
@@ -0,0 +1,216 @@
+        render_phase::RenderPhase,
+        render_resource::{
+            Extent3d, TextureDescriptor, TextureDimension, TextureFormat, TextureUsages,
+        },
+        renderer::RenderContext,
+        view::RenderLayers,
+        RenderApp, RenderStage,
+    },
+};
+
+pub const RENDER_IMAGE_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Image::TYPE_UUID, 13378939762009864029);
+
+pub const FIRST_PASS_DRIVER: &str = "first_pass_driver";
+pub const FIRST_PASS_CAMERA: &str = "first_pass_camera";
+
+#[derive(Component)]
+struct FirstPassCube;
+#[derive(Component)]
+struct MainPassCube;
+
+/// rotates the inner cube (first pass)
+fn rotator_system(time: Res<Time>, mut query: Query<&mut Transform, With<FirstPassCube>>) {
+    for mut transform in query.iter_mut() {
+        transform.rotation *= Quat::from_rotation_x(1.5 * time.delta_seconds());
+        transform.rotation *= Quat::from_rotation_z(1.3 * time.delta_seconds());
+    }
+}
+
+/// rotates the outer cube (main pass)
+fn cube_rotator_system(time: Res<Time>, mut query: Query<&mut Transform, With<MainPassCube>>) {
+    for mut transform in query.iter_mut() {
+        transform.rotation *= Quat::from_rotation_x(1.0 * time.delta_seconds());
+        transform.rotation *= Quat::from_rotation_y(0.7 * time.delta_seconds());
+    }
+}
+
+fn setup(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut active_cameras: ResMut<ActiveCameras>,
+    mut images: ResMut<Assets<Image>>,
+    mut clear_color: ResMut<ClearColor>,
+) {
+    let size = Extent3d {
+        width: 512,
+        height: 512,
+        ..Default::default()
+    };
+    let image = Image {
+        data: vec![0; size.width as usize * size.height as usize * 4],
+        texture_descriptor: TextureDescriptor {
+            label: None,
+            size,
+            dimension: TextureDimension::D2,
+            format: TextureFormat::Bgra8UnormSrgb,
+            mip_level_count: 1,
+            sample_count: 1,
+            usage: TextureUsages::TEXTURE_BINDING
+                | TextureUsages::COPY_DST
+                | TextureUsages::RENDER_ATTACHMENT,
+        },
+        ..Default::default()
+    };
+    let image_handle = images.set(RENDER_IMAGE_HANDLE, image);
+
+    let cube_handle = meshes.add(Mesh::from(shape::Cube { size: 4.0 }));
+    let cube_material_handle = materials.add(StandardMaterial {
+        base_color: Color::rgb(0.8, 0.7, 0.6),
+        reflectance: 0.02,
+        unlit: false,
+        ..Default::default()
+    });
+
+    commands
+        .spawn_bundle(PbrBundle {
+            mesh: cube_handle,
+            material: cube_material_handle,
+            transform: Transform::from_translation(Vec3::new(0.0, 0.0, 1.0)),
+            ..Default::default()
+        })
+        .insert(FirstPassCube)
+        .insert(RenderLayers::layer(1));
+
+    // light
+    // note: currently lights are shared between passes!
```

Comment:
File an issue for this please?

Reply:
Done: https://github.com/bevyengine/bevy/issues/3462